### PR TITLE
Prevent mobile view change on location clear

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
Currently when you clear a location field in mobile view, it resets the view back to the form. This PR changes that so now using the clear location button just clears the location.

This PR also updates the mobile bar back button to be web accessible, and turns it on for the location field. 

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
I think we do need a close or back button here. The browser back button doesn't work to go back to the trip form. :( Sorry

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Ooo yes nicely done! Thanks!

**Comment by binh-dam-ibigroup** [comment]
@amy-corson-ibigroup Do we need to fix some of the Percy tests?

**Comment by amy-corson-ibigroup** [comment]
@binh-dam-ibigroup Fixed!

**Comment by amy-corson-ibigroup** [comment]
As @binh-dam-ibigroup pointed out, the mobile nav bar already has a back arrow, so I've used that instead of creating my own! I've also updated the back arrow to be web accessible but changing it from a `div` to a `button` and adding an `aria-label`

File: lib/components/mobile/mobile.css [hunk 1]
```diff
@@ -52,9 +52,10 @@
 }
 
 .otp .navbar .mobile-back {
+  background: transparent;
+  border: none;
   position: fixed;
-  top: 12px;
-  left: 15px;
+  left: 7px;
   color: white;
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/mobile.css:59]: Sort CSS

File: lib/components/mobile/navigation-bar.js [hunk 2]
```diff
@@ -51,21 +53,28 @@ class MobileNavigationBar extends Component {
...
+    const backButtonText = intl.formatMessage({ id: 'common.forms.back' })
+
     return (
       <header>
         <Navbar className="mobile-navbar-container" fixedTop fluid>
           <Navbar.Header>
             <Navbar.Brand>
               {showBackButton ? (
-                <div className="mobile-back">
+                <button
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/navigation-bar.js:69]: Nice

**Review by binh-dam-ibigroup — APPROVED** [review]
> As @binh-dam-ibigroup pointed out, the mobile nav bar already has a back arrow, so I've used that instead of creating my own! I've also updated the back arrow to be web accessible but changing it from a `div` to a `button` and adding an `aria-label`

Nice fix!

**Pull request merged**